src/Giraffe/Serialization.fs

```diff
@@ -92,11 +94,11 @@ module Json =
                 |> Encoding.UTF8.GetBytes
 
             member __.SerializeToStreamAsync (x : 'T) (stream : Stream) =
-                use sw = new StreamWriter(stream, Utf8EncodingWithoutBom, DefaultBufferSize, true)
-                use jw = new JsonTextWriter(sw)
-                let sr = JsonSerializer.Create settings
-                sr.Serialize(jw, x)
-                Task.CompletedTask
+                let memoryStream = new MemoryStream()
+                let streamWriter = new StreamWriter(memoryStream, Utf8EncodingWithoutBom)
```

Comment:
Should we be using the `use` keyword for the `StreamWriter` as it is disposable (same for the `JsonTextWriter` and the readers down below)?

Reply:
I was sure this is for disposal of underlying stream, in this case the stream is a resize array, but I will check

Reply:
Good catch, it flushes buffer on dispose, will introduce `use`s where they were

Reply:
Do those flushes need to be async too?  See here: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#always-call-flushasync-on-streamwriters-or-streams-before-calling-dispose

Reply:
Yeah, needs to be adressed, great catch!

Reply:
There is no need in calling flush async here because writes to memory stream are synchronous anyway. 
I've added manual flush before copy to the output though

Reply:
Cool, I thought that might be the case, I sometimes get in a middle about what should be flushed when, especially when there are multiple streams involved